bluemira/utilities/tools.py

```diff
@@ -302,6 +303,28 @@ def cross(
 # =====================================================
 
 
+class ColourDescriptor:
+    """Colour Descriptor for use with dataclasses"""
+
+    def __init__(self):
+        self._default = colors.to_hex((0.5, 0.5, 0.5))
+
+    def __set_name__(self, _, name: str):
+        """Set the attribute name from a dataclass"""
+        self._name = "_" + name
+
+    def __get__(self, obj: Any, _) -> str:
+        """Get the hex colour"""
+        if obj is None:
+            return self._default
+
+        return colors.to_hex(getattr(obj, self._name, self._default))
+
+    def __set__(self, obj: Any, value: Union[str, Tuple[float, ...]]):
```

Comment:
Are we restricted to RGB? In which case, shouldn't this be `Tuple[float, float, float]`?

Reply:
Is that potentially worth adding to the docstring?